index.d.ts

```diff
@@ -0,0 +1,611 @@
+/* tslint:disable:interface-name max-classes-per-file no-empty-interface */
+
+declare module 'evergreen-ui' {
+  import * as React from 'react'
+  import { EnhancerProps } from 'ui-box/dist/types/enhancers'
+
+  interface UIBoxProps extends Partial<EnhancerProps> {
+    innerRef?: React.Ref<HTMLElement>;
+    children?: React.ReactNode;
+    height?: string | number;
+    width?: string | number;
+    css?: object;
+  }
+
+  type PositionTypes = 'top' | 'top-left' | 'top-right' | 'bottom' | 'bottom-left' | 'bottom-right' | 'left' | 'right';
+  type IntentTypes = 'none' | 'success' | 'warning' | 'danger';
+  type IconNameTypes =
+    ''
+    | 'loading'
+    | 'error'
+    | 'tick-circle'
+    | 'caret-down'
+    | 'cross'
+    | 'menu'
+    | 'list'
+    | 'more'
+    | 'properties'
+    | 'diagram-tree'
+    | 'database'
+    | 'join-table'
+    | 'function'
+    | 'person'
+    | 'import'
+    | 'predictive-analysis'
+    | 'refresh'
+    | 'export';
+
+  type RemovedAttributes = 'color' | 'title' ;
+  type ElementProps = Omit<React.HtmlHTMLAttributes<HTMLElement>, RemovedAttributes>;
+  type ElementInputProps = Omit<React.InputHTMLAttributes<HTMLInputElement>, RemovedAttributes>;
+  type ElementButtonProps = Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, RemovedAttributes>;
```

Comment:
Why do we need to omit these props?

Also can we write this as `Omit<React.HTMLAttributes<HTMLElement>, RemovedAttributes>` – just not sure what `HtmlHTMLAttributes` gets us (it adds an optional `manifest` attribute...)

Reply:
Curious if it would be better for us to omit these properties only on the components that expose those same prop names?

Reply:
This Props and types was a work around for various React/ui-box props which conflicts with component props. If the use `BoxProps<T>` then ref still broken, but conflicts will be avoided and props a no longer nessary.
Unless you can provide better sniped then i can figured out